test/functional/terminal/buffer_spec.lua

```diff
@@ -255,8 +260,22 @@ describe(':terminal buffer', function()
     command('bdelete!')
   end)
 
-  it('handles wqall', function()
+  it("requires bang (!) to close a running job", function()
+    local cwd = funcs.fnamemodify('.', ':p:~'):gsub([[[\/]*$]], '')
+    local ext_pat = iswin() and '%.EXE' or ''
     eq('Vim(wqall):E948: Job still running', exc_exec('wqall'))
+    matches('^Vim%(bdelete%):E89: term://'..pesc(cwd)..'//%d+:'..nvim_dir..'/tty%-test'..ext_pat..' will be killed %(add %! to override%)$', exc_exec('bdelete'))
```

Comment:
- we could simplify the test by not being so precise. I don't think we lose much by changing the pattern to ignore the CWD and EXE. Up to you, though.
    - this anticipates a glorious future where we change the `term://` format https://github.com/neovim/neovim/issues/3278

```suggestion
    matches('^Vim%(bdelete%):E89: term://.*tty%-test.* will be killed %(add %! to override%)$', exc_exec('bdelete'))
```

Reply:
No objection to this.